overlord/snapstate/agentnotify/agentnotify_test.go

```diff
@@ -0,0 +1,85 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package agentnotify_test
+
+import (
+	"testing"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/snapstate/agentnotify"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+)
+
+func TestAgentNotify(t *testing.T) { TestingT(t) }
+
+type agentNotifySuite struct {
+	st *state.State
+}
+
+var _ = Suite(&agentNotifySuite{})
+
+func (s *agentNotifySuite) SetUpTest(c *C) {
+	s.st = state.New(nil)
+
+}
+
+func (s *agentNotifySuite) TestNotifyAgentOnLinkChange(c *C) {
+	s.st.Lock()
+	defer s.st.Unlock()
+
+	var i int
```

Comment:
nitpick: since this variable's scope is not that small, I'd give a larger name like `callCount`